src/opencensus/proto/agent/README.md

```diff
@@ -57,6 +57,58 @@ are established.
 expired on Agent side. Library needs to start a new connection with a unique identifier
 (MAY be different than the previous one).
 
+## Implementation details of Agent Server
+
+This section describes the in-process implementation details of OC-Agent.
+
+![agent-implementation](agent-implementation.png)
+
+The Agent consists of three main parts:
+
+1. The interceptors of different instrumentation libraries, such as OpenCensus, Zipkin,
+Istio Mixer, Prometheus client, etc. Interceptors act as the “frontend” or “gateway” of
+Agent. In addition, there MAY be one special receiver for receiving configuration updates
+from outside.
+2. The core Agent module. It acts as the “brain” or “dispatcher” of Agent.
+3. The exporters to different monitoring backends or collector services, such as
+Omnition Collector, Stackdriver Trace, Jaeger, Zipkin, etc.
+
+### Interceptors
+
+Each interceptor can be connected with multiple instrumentation libraries. The
+communication protocol between interceptors and libraries is the one we described in the
+proto files (for example trace_service.proto). When a library opens the connection with the
+corresponding interceptor, the first message it sends must have the `Node` identifier. The
+interceptor will then cache the `Node` for each library, and `Node` is not required for
+the subsequent messages from libraries.
+
+### Agent Core
+
+Most functionalities of Agent are in Agent Core. Agent Core's responsibilies include:
+
+1. Accept `SpanProto` from each interceptor. Note that the `SpanProto`s that are sent to
```

Comment:
This makes it seem like we need to send `Node` for every `SpanProto`. How about
```
1. Accept SpanProto from each interceptor. Note that just like the rest of the initiating protocols, the first message that should have been sent by each interceptor MUST have Node set; Node is then cached by Agent and subsequent
SpanProto messages sent by the interceptor don't need Node. The associated Node is neccessary so that Agent core can differentiate and group SpanProto's.
```

Reply:
IIUC the API for accepting `SpanProto` on Agent Core is stateless, so indeed the interceptor needs to send `Node` for every `SpanProto` (or batched `SpanProto`s). This all happens within the process, so we don't need to worry about the encoded size issue.

/cc @bogdandrutu

Reply:
In that case I'll defer to Bogdan to do the approval as I am only learning about the interceptors and other part right from this review.